ddtrace/propagation/http.py

```diff
@@ -678,6 +678,16 @@ def _extract(headers):
             meta=meta,
         )
 
+    @staticmethod
+    def _inject(span_context, headers):
+        # type: (Context, Dict[str, str]) -> None
+        tp = span_context._traceparent
+        if tp:
+            headers[_HTTP_HEADER_TRACEPARENT] = tp
+        ts = span_context._tracestate
+        if ts:
```

Comment:
nit: according to the w3c spec trace state should only be set if a trace parent is also set: https://www.w3.org/TR/trace-context/#tracestate-header. We either return early if tp is None/empty string or add a check here

Reply:
ah good catch, thank you

Reply:
I'll add a test for this as well